xds/src/main/java/io/grpc/xds/WeightedTargetLoadBalancer.java

```diff
@@ -131,20 +135,28 @@ private void updateOverallBalancingState() {
     List<WeightedChildPicker> childPickers = new ArrayList<>();
 
     ConnectivityState overallState = null;
+    Status errorStatus = Status.UNAVAILABLE;
     for (String name : targets.keySet()) {
       ChildHelper childHelper = childHelpers.get(name);
       ConnectivityState childState = childHelper.currentState;
       overallState = aggregateState(overallState, childState);
       if (READY == childState) {
         int weight = targets.get(name).weight;
         childPickers.add(new WeightedChildPicker(weight, childHelper.currentPicker));
+      } else if (TRANSIENT_FAILURE == overallState) {
+        // do a pick to get more failure details
+        PickSubchannelArgs args = new PickSubchannelArgsImpl(
+            getStreamAggregatedResourcesMethod(),
+            new Metadata(),
+            CallOptions.DEFAULT.withOption(XdsNameResolver.CLUSTER_SELECTION_KEY, name));
+        errorStatus = childHelper.currentPicker.pickSubchannel(args).getStatus();
```

Comment:
I don't think we should save the status like this. It seems like it could easily be out-of-date, and we have to manufacture an RPC pick to trigger it which is asking for bugs, as higher-level policies wouldn't be able to modify this pick.

It would be much better to use the failing picker directly instead of `ErrorPicker`. Although, I still think it'd be even better to use WeightedRandomPicker with all the failing transports. We could have a `List<WeightedChildPicker> errorChildPickers` and use it to create WeightedRandomPicker if `overallState == TRANSIENT_FAILURE`.